index.js

```diff
@@ -168,18 +168,28 @@ function main() {
  */
 function getNativeBinary() {
   let nativeBinary;
-
-  if (os.platform() === 'win32') {
+  const platform = os.platform();
+  const arch = os.arch();
+  if (platform === 'win32') {
     nativeBinary = `${__dirname}/bin/win32/clang-format.exe`;
   } else {
-    nativeBinary = `${__dirname}/bin/${os.platform()}_${os.arch()}/clang-format`;
+    nativeBinary = `${__dirname}/bin/${platform}_${arch}/clang-format`;
   }
 
   if (!fs.existsSync(nativeBinary)) {
     const message = 'This module doesn\'t bundle the clang-format executable for your platform. ' +
-        `(${os.platform()}_${os.arch()})\n` +
+        `(${platform}_${arch})\n` +
         'Consider installing it with your native package manager instead.\n';
-    throw new Error(message);
+    if (platform === 'darwin' && arch === 'arm64') {
```

Comment:
maybe we could write this a little less repetitively?

E.g.

```
if (existsSync(nativeBinary)) return nativeBinary;
if (darwin && arm64) { check if darwin_x64 exists, return if so }
construct error message and throw
```

That'd avoid the duplicate throw, and also move all error msg construction and handling after the positive paths.
```

Reply:
@mprobst Ok, I've force-pushed a new commit which is a little cleaner and follows the early return suggestion here.